data_structures/src/radon_report.rs

```diff
@@ -17,6 +17,8 @@ where
 {
     /// Stage-specific metadata.
     pub metadata: Stage,
+    /// Vector of partial results (the results in between each of the operators in a script)
+    pub partial_results: Vec<RT>,
```

Comment:
Could we make this field optional? If we always store the partial results we will be using extra memory for no reason, because we don't need partial results when executing data requests in the node, right?

I would keep the old behavior and add a new method `execute_radon_script_with_partial_results` or similar, and maybe even pass `partial_results: &mut Vec<_>` as an input argument, to avoid having to handle an optional field in `RadonReport`, if that doesn't complicate thinks in the wallet API.

Reply:
I made some changes in that line, but through an approach that I believe that will give us better extensibility in the future. The extension of the changes is a little bigger than I wanted, but at least it has zero impact to any existing APIs.